src/Microsoft.PowerShell.ConsoleHost/host/msh/ConsoleHost.cs

```diff
@@ -1752,6 +1752,33 @@ private void DoRunspaceInitialization(bool skipProfiles, string initialCommand,
             TelemetryAPI.ReportStartupTelemetry(this);
 #endif
 
+            // If working directory was specified, set it
+            if (s_cpp != null && s_cpp.WorkingDirectory != null)
+            {
+                Pipeline tempPipeline = exec.CreatePipeline();
+                Command c = new Command("Set-Location");
+                c.Parameters.Add("LiteralPath", s_cpp.WorkingDirectory);
+                tempPipeline.Commands.Add(c);
+
+                Exception e1;
```

Comment:
C#7 should allow you to inline the declaration, when you call the methods below:
````powershell
if (IsRunningAsync)
{
    exec.ExecuteCommandAsyncHelper(tempPipeline, out Exception e1, Executor.ExecutionOptions.AddOutputter);
}
else
{
    exec.ExecuteCommandHelper(tempPipeline, out Exception e1, Executor.ExecutionOptions.AddOutputter);
}
````

Reply:
Will update

Reply:
Not able to use this syntax as e1 is used outside the scope of the braces

Reply:
Ah, I see. Sorry. I thought because it is an if/else expression, the compiler is intelligent enough to know that `e1` will always be declared.